File: session 6/concurrency.py
import threading
import time

def task(name):
    for i in range(5):
        print(f"Task {name} is running iteration {i}")
        time.sleep(1)  # Simulate work

# Create threads
thread1 = threading.Thread(target=task, args=("A",))
thread2 = threading.Thread(target=task, args=("B",))

# Start threads
thread1.start()
thread2.start()

# Wait for both threads to complete
thread1.join()
thread2.join()

print("All tasks completed")

File: session 6/concurrency.go
package main

import (
	"fmt"
	"time"
)

func task(name string) {
	for i := 0; i < 5; i++ {
		fmt.Printf("Task %s is running iteration %d\n", name, i)
		time.Sleep(1 * time.Second) // Simulate work
	}
}

func main() {
	go task("A") // Start a goroutine for task A
	go task("B") // Start a goroutine for task B

	time.Sleep(6 * time.Second) // Wait for the tasks to finish
	fmt.Println("All tasks completed")
}
